server/plugin/api.go

```diff
@@ -555,20 +556,20 @@ func (p *Plugin) getConnected(c *Context, w http.ResponseWriter, r *http.Request
 	config := p.getConfiguration()
 
 	type ConnectedResponse struct {
-		Connected         bool               `json:"connected"`
-		GitHubUsername    string             `json:"github_username"`
-		GitHubClientID    string             `json:"github_client_id"`
-		EnterpriseBaseURL string             `json:"enterprise_base_url,omitempty"`
-		Organization      string             `json:"organization"`
-		UserSettings      *UserSettings      `json:"user_settings"`
-		PluginSettings    ClientSafeSettings `json:"plugin_settings"`
+		Connected           bool                   `json:"connected"`
+		GitHubUsername      string                 `json:"github_username"`
+		GitHubClientID      string                 `json:"github_client_id"`
+		EnterpriseBaseURL   string                 `json:"enterprise_base_url,omitempty"`
+		Organization        string                 `json:"organization"`
+		UserSettings        *UserSettings          `json:"user_settings"`
+		ClientConfiguration map[string]interface{} `json:"configuration"`
```

Comment:
Why is gorp able to process `*UserSettings` but not `ClientSafeSettings`?

Reply:
`ConnectedResponse`, which contains `UserSettings` is send a json response, not via websocket. `ClientConfiguration` could be a `struct` here. But not in https://github.com/mattermost/mattermost-plugin-github/blob/e5ff23f189aa15fe2d6bbdd0b325edfb79d91457/server/plugin/api.go#L479-L490 where it's send as a websocket payload.